synapse/config/auth.py

```diff
@@ -29,7 +29,15 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
         if password_config is None:
             password_config = {}
 
-        self.password_enabled = password_config.get("enabled", True)
+        passwords_enabled = password_config.get("enabled", True)
+        self.password_enabled = passwords_enabled
+        # 'only_for_reauth' allows users who have previously set a password to use it,
+        # even though passwords would otherwise be disabled.
+        self.password_enabled_for_reauth = passwords_enabled == "only_for_reauth"
```

Comment:
shouldn't this be true if `password_enabled` is true? If passwords are enabled in general, then they are also enabled for reauth?